x-pack/plugins/transform/server/routes/api/error_utils.ts

```diff
@@ -83,3 +83,58 @@ export function wrapError(error: any): CustomHttpResponseOptions<ResponseError>
     statusCode: boom.output.statusCode,
   };
 }
+
+function extractCausedByChain(
+  causedBy: Record<string, any> = {},
+  accumulator: string[] = []
+): string[] {
+  const { reason, caused_by } = causedBy; // eslint-disable-line @typescript-eslint/naming-convention
+
+  if (reason) {
+    accumulator.push(reason);
+  }
+
+  if (caused_by) {
+    return extractCausedByChain(caused_by, accumulator);
+  }
+
+  return accumulator;
+}
+
+/**
+ * Wraps an error thrown by the ES JS client into a Boom error response and returns it
+ *
+ * @param err Object Error thrown by ES JS client
+ * @param statusCodeToMessageMap Object Optional map of HTTP status codes => error messages
+ * @return Object Boom error response
+ */
+export function wrapEsError(err: any, statusCodeToMessageMap: Record<string, any> = {}) {
+  const { statusCode, response } = err;
+
+  const {
+    error: {
+      root_cause = [], // eslint-disable-line @typescript-eslint/naming-convention
+      caused_by = {}, // eslint-disable-line @typescript-eslint/naming-convention
+    } = {},
+  } = JSON.parse(response);
```

Comment:
@elastic/ml-ui The `x-pack/plugins/transform/server/routes/api` module was the last and only consumers of `wrapEsError` which was previously in `x-pack/legacy/server/lib/create_router/error_wrappers/wrap_es_error.js`. I moved the implementation here, in the same file as `wrapError`

Reply:
Thanks for the heads up!